### ISSUE TITLE
tensorflow:Error recorded from training_loop

### ISSUE THREAD

**Issue by skyornig** [zhp510730568/bert-ad#1]
I am running run_pretraining_mgpu.py with n_gpus=2, and I got the error info below:

2020-05-29 11:17:55.751296: W tensorflow/core/common_runtime/bfc_allocator.cc:319] ****************************************************************************************************
ERROR:tensorflow:Error recorded from training_loop: 2 root error(s) found.
  (0) Internal: Dst tensor is not initialized.
         [[node replica_1/gradients/replica_1/bert/encoder/layer_10/attention/self/value/MatMul_grad/tuple/control_dependency_1 (defined at run_pretraining_mgpu.py:334) ]]
  (1) Internal: Dst tensor is not initialized.
         [[node replica_1/gradients/replica_1/bert/encoder/layer_10/attention/self/value/MatMul_grad/tuple/control_dependency_1 (defined at run_pretraining_mgpu.py:334) ]]
         [[replica_0/add_2/_6337]]
0 successful operations.
1 derived errors ignored.

Original stack trace for 'replica_1/gradients/replica_1/bert/encoder/layer_10/attention/self/value/MatMul_grad/tuple/control_dependency_1':
  File "run_pretraining_mgpu.py", line 675, in <module>
    tf.app.run()
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 40, in run
    _run(main=main, argv=argv, flags_parser=_parse_flags_tolerate_undef)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "run_pretraining_mgpu.py", line 648, in main
    estimator.train(input_fn=train_input_fn, max_steps=FLAGS.num_train_steps)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 2871, in train
    saving_listeners=saving_listeners)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/estimator.py", line 367, in train
    loss = self._train_model(input_fn, hooks, saving_listeners)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/estimator.py", line 1158, in _train_model
    return self._train_model_default(input_fn, hooks, saving_listeners)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/estimator.py", line 1188, in _train_model_default
    features, labels, ModeKeys.TRAIN, self.config)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 2709, in _call_model_fn
    config)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/estimator.py", line 1146, in _call_model_fn
    model_fn_results = self._model_fn(features=features, **kwargs)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 2967, in _model_fn
    features, labels, is_export_mode=is_export_mode)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 1549, in call_without_tpu
    return self._call_model_fn(features, labels, is_export_mode=is_export_mode)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py", line 1867, in _call_model_fn
    estimator_spec = self._model_fn(features=features, **kwargs)
  File "run_pretraining_mgpu.py", line 334, in model_fn
    aggregation_method=tf.AggregationMethod.EXPERIMENTAL_TREE,
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow/python/training/optimizer.py", line 512, in compute_gradients
    colocate_gradients_with_ops=colocate_gradients_with_ops)
  File "/public/zxf/anaconda3/envs/nlp_py36_tfGPU_1.14/lib/python3.6/site-packages/tensorflow/python/ops/gradients_impl.py", line 158, in gradients
    unconnected_gradients)

Could I need make some more changes to run pretraining code?
